### PR TITLE
Debug new ephy_testing_data

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Recent changes to the ephy_testing_data have caused some data tests to start failing. Investigating causes and workarounds now (suspect it is an improperly stubbed file).

### CODE REVIEW

File: tests/test_on_data/test_metadata/test_spikeglx_metadata.py [hunk 1]
```diff
@@ -76,10 +77,11 @@ def test_spikelgx_recording_property_addition():
     assert_array_equal(contact_shapes, expected_contact_shapes)
 
 
+@pytest.mark.skip(reason="Legacy spikeextractors cannot read new GIN file.")
 def test_matching_recording_property_addition_between_backends():
     """Test that the extracted properties match with both backends"""
-    folder_path = SPIKEGLX_PATH / "Noise4Sam_g0" / "Noise4Sam_g0_imec0"
-    ap_file_path = folder_path / "Noise4Sam_g0_t0.imec0.ap.bin"
+    folder_path = SPIKEGLX_PATH / "TEST_20210920_0_g0"
+    ap_file_path = folder_path / "TEST_20210920_0_g0_t0.imec0.ap.bin"
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply tests/test_on_data/test_metadata/test_spikeglx_metadata.py:84]: They're both comparably small now, I suppose this part can be kept.

File: tests/test_on_data/test_gin_ecephys.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ class TestConverter(NWBConverter):
                 )
             )
 
-    for spikeextractors_backend in [True, False]:
+    for spikeextractors_backend in [False]:  # Cannot run try since legacy spikeextractors cannot read new GIN file
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply tests/test_on_data/test_gin_ecephys.py:133]: I'd rather add another properly stubbed SpikeGLX file, personally.

**Review by h-mayorquin — APPROVED** [review]

**Comment by samuelgarcia** [comment]
Maybe it is a issue with the fact that the file size have to be a multiple of channel (384 + 1)

This line https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extractors/spikeglxrecordingextractor/spikeglxrecordingextractor.py#L63
create a memmap without taking care that the number of sample is compatible with the number of channel.

neo do exatly the same : https://github.com/NeuralEnsemble/python-neo/blob/master/neo/rawio/spikeglxrawio.py#L96
I will fix the neo implementation for this.

Does it also bug with neo + new SI ?

**Comment by CodyCBakerPhD** [comment]
@samuelgarcia 

> Maybe it is a issue with the fact that the file size have to be a multiple of channel (384 + 1)

I also figured it had something to do with that, since the number of channels is specified in the header and the header value used in the old version.

The files are still loadable via SpikeInterface, likely due to that `-1` open slicing on the reshape. Though I did not thoroughly check if the return structure from `get_traces` matches what it was before the new stubbing.

**Pull request merged**